src/imitation/util/logger.py

```diff
@@ -0,0 +1,126 @@
+import contextlib
+import os
+from typing import Dict, Optional, Sequence
+
+from stable_baselines.common.misc_util import mpi_rank_or_zero
+import stable_baselines.logger as sb_logger
+
+
+def _sb_logger_configure_replacement(*args, **kwargs):
+  raise RuntimeError("Shouldn't call stable_baselines.logger.configure "
+                     "once imitation.logger.configure() has been called")
+
+
+def _sb_logger_reset_replacement():
+  raise NotImplementedError("reset() while using imitation.configure()")
+
+
+_in_accumul_context = False
```

Comment:
Consider encapsulating state in a class rather than as globals. I think this would make things easier to read, and would also support the use case of having multiple of these imitation-specific loggers (not sure when we'd need this yet but may come up).

How about making a new class e.g. `HierarchicalLogger` that has e.g. `self.current : Logger` and `self.cached_logger` attributes and implements the same abstract interface as `Logger`? Methods like `logkv` etc you'd just forward to `self.current`. You could then make `accumulate_means` an instance method that switches `self.current`.

`configure()` could then set an instance of this class as `sb_logger.Logger.CURRENT` and you wouldn't have to update this again.

Reply:
Nice, that does seem like it would be cleaner.

Reply:
@AdamGleave Why would we need to concatenate data from the raw CSVs together? Isn't the main point of having the means logging so that we can sidestep comparing raw logs with conflicting timesteps?

Reply:
done.

Reply:
I made a `HierarchicalLogger` that I think mostly does what you described. I struggled for a little bit figuring out where to put the formerly global `_format_strs` variable (needed to make sure that logging outputs are consistent with the default logger when we switch to "raw" loggers). Turns out that format_str fits pretty well as a local variable within HierarchicalLogger, especially if HierarchicalLogger instantiates the new AccumulatingLogger.